AK/JsonArraySerializer.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <AK/JsonValue.h>
+
+namespace AK {
+
+template<typename Builder>
+class JsonObjectSerializer;
+
+template<typename Builder>
+class JsonArraySerializer {
+public:
+    JsonArraySerializer(Builder& builder)
+        : m_builder(builder)
+    {
+        m_builder.append('[');
+    }
+
+    JsonArraySerializer(const JsonArraySerializer&) = delete;
+    JsonArraySerializer(JsonArraySerializer&&) = delete;
+
+    ~JsonArraySerializer()
+    {
+        if (!m_finished)
+            finish();
+    }
+
+    void add(const JsonValue& value)
+    {
+        begin_item();
+        value.serialize(m_builder);
+    }
+
+    JsonArraySerializer<Builder> add_array()
```

Comment:
Let's add `[[nodiscard]]` this so the compiler warns you if you call it without using the return value.

Reply:
Why, maybe you want to add an empty array

Reply:
Hmm.. You know what.. I guess maybe you would want to do that. Never mind about `[[nodiscard]]`. :)